daemon/watchdog_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package daemon
+
+import (
+	"os"
+	"strconv"
+	"testing"
+)
+
+func TestSdWatchdogEnabled(t *testing.T) {
+	// (time, nil)
+	err := os.Setenv("WATCHDOG_USEC", "100")
+	if err != nil {
+		panic(err)
```

Comment:
I actually disagree on that one - my usual classification is:
- `t.Error` for test failures related to user code that can continue
- `t.Fatal` for test failures related to user code that cannot continue
- `panic` for test failures unrelated to user code e.g. in setup etc (very explicit that it's an environment failure/solar flare and not a code bug)